libs/git/repository.go

```diff
@@ -23,8 +23,18 @@ type Repository struct {
 	// directory where we process .gitignore files.
 	real bool
 
-	// root is the absolute path to the repository root.
-	root vfs.Path
+	// checkoutDir is the path to the root of the repository checkout.
+	// This can be either the main repository checkout or a worktree.
+	checkoutDir vfs.Path
+
+	// gitDir is the equivalent of $GIT_DIR and points to the
+	// `.git` directory of a repository or a worktree.
+	gitDir vfs.Path
+
+	// gitCommonDir is the equivalent of $GIT_COMMON_DIR and points to the
+	// `.git` directory of the main working tree (common between worktrees).
+	// This is equivalent to [gitDir] if this is the main working tree.
+	gitCommonDir vfs.Path
```

Comment:
Please include URL for docs explaining the difference between `GIT_DIR` and `GIT_COMMON_DIR` in this context:
https://git-scm.com/docs/git-worktree#_details

Reply:
Added the link to the comment for both fields.